diff --git box.h box.h
index 0d053a4..c536172 100644
--- box.h
+++ box.h
@@ -1300,6 +1300,7 @@ typedef struct
     isom_dinf_t *dinf;      /* ISOM: Data Information Box / QTFF: null */
     isom_keys_t *keys;      /* ISOM: null / QTFF: Metadata Item Keys Box */
     isom_ilst_t *ilst;      /* Metadata Item List Box only defined in Apple MPEG-4 and QTFF */
+    isom_free_t *free;      /* Padding */
 } isom_meta_t;
 
 /* Window Location Box */
diff --git isom.c isom.c
index 4ea1e43..355ae96 100644
--- isom.c
+++ isom.c
@@ -1611,6 +1611,16 @@ int isom_add_meta( isom_box_t *parent )
             return -1;
         }
         udta->meta = meta;
+        isom_create_box( skip, parent, ISOM_BOX_TYPE_FREE );
+        if( meta->free )
+        {
+            free( skip );
+            return -1;
+        }
+        meta->free = skip;
+        skip->data = calloc(1, 4096);
+        skip->length = 4096;
+        skip->size = 8 + skip->length;
     }
     return 0;
 }
@@ -2597,6 +2607,11 @@ static void isom_remove_meta( isom_meta_t *meta )
     isom_remove_dinf( meta->dinf );
     isom_remove_keys( meta->keys );
     isom_remove_ilst( meta->ilst );
+    if( meta->free) {
+        free( meta->free->data );
+        free( meta->free );
+        meta->free = NULL;
+    }
     if( meta->parent )
     {
         if( lsmash_check_box_type_identical( meta->parent->type, LSMASH_BOX_TYPE_UNSPECIFIED ) )
@@ -4521,6 +4536,7 @@ static uint64_t isom_update_meta_size( isom_meta_t *meta )
         + isom_update_hdlr_size( meta->hdlr )
         + isom_update_dinf_size( meta->dinf )
         + isom_update_ilst_size( meta->ilst );
+    if( meta->free ) meta->size += meta->free->size;
     CHECK_LARGESIZE( meta );
     return meta->size;
 }
diff --git meta.c meta.c
index 54c8f2e..2c5a602 100644
--- meta.c
+++ meta.c
@@ -178,7 +178,8 @@ static int isom_set_itunes_metadata_binary( lsmash_root_t *root,
         case ITUNES_METADATA_ITEM_COVER_ART :
             if( value.binary.subtype != ITUNES_METADATA_SUBTYPE_JPEG
              && value.binary.subtype != ITUNES_METADATA_SUBTYPE_PNG
-             && value.binary.subtype != ITUNES_METADATA_SUBTYPE_BMP )
+             && value.binary.subtype != ITUNES_METADATA_SUBTYPE_BMP
+             && value.binary.subtype != ITUNES_METADATA_SUBTYPE_GIF )
                 return -1;
             break;
         case ITUNES_METADATA_ITEM_DISC_NUMBER :
diff --git write.c write.c
index 96dbc0b..1a85e89 100644
--- write.c
+++ write.c
@@ -1457,6 +1457,12 @@ int isom_write_meta( lsmash_bs_t *bs, isom_meta_t *meta )
      || isom_write_dinf( bs, meta->dinf, meta->type )
      || isom_write_ilst( bs, meta->ilst ) )
         return -1;
+    if( meta->free) {
+       isom_bs_put_box_common( bs, meta->free );
+        if( meta->free->data && meta->free->length )
+            lsmash_bs_put_bytes( bs, meta->free->length, meta->free->data );
+        lsmash_bs_write_data( bs );
+    }
     return 0;
 }
 
